storage/innobase/fil/fil0crypt.cc

```diff
@@ -2523,17 +2482,20 @@ static void fil_crypt_complete_rotate_space(const key_state_t *key_state,
       } else {
         mutex_enter(&crypt_data->mutex);
         crypt_data->rotate_state.flushing = false;
-        ib::error() << "Encryption thread failed to flush encryption "
-                       "information for tablespace "
-                    << state->space->name
-                    << ". This should not happen and could indicate problem "
-                       "with OS or filesystem. Excluding "
-                    << state->space->name << " from encryption rotation. "
-                    << "You can try decrypting/encrypting with alter statement "
-                       "for this table or restarting the server.";
-        state->space->exclude_from_rotation =
-            true;  // This will stop encryption threads from picking up this
-                   // tablespace for rotation
+        if (!state->space->stop_new_ops) {  // Flag updated failed not due to
```

Comment:
I think this shouldn't happen. Or there should be better to way handle this. See if any of the below idea works.

1. increase fil_space_t ref_count and concurrent drop is prevented.
2. See if MDL shared lock can be acquired and that can prevent concurrent DROP.

Reply:
This can happen if we failed to acquired S lock on a table because drop has acquired X lock on it beforehand.